src/librustc/mir/interpret/eval_context.rs

```diff
@@ -413,7 +413,7 @@ impl<'a, 'tcx, M: Machine<'tcx>> EvalContext<'a, 'tcx, M> {
 
     /// Returns the normalized type of a struct field
     fn field_ty(&self, param_substs: &Substs<'tcx>, f: &ty::FieldDef) -> ty::Ty<'tcx> {
```

Comment:
`field_ty` should be replaced with `.field(...).ty` on a `TyLayout` (and if the type was used to obtain a layout, that means only `.field(...)` instead of `field_ty` followed by `layout_of`).

Reply:
That would need a big refactoring in so many places... It's just a performance thing, right?

Reply:
No, it's an API thing, but probably the best to wait for #45225.